secure_enclave/TEUtils.cpp

```diff
@@ -65,152 +70,150 @@ std::string fieldElementToString(const T &field_elem, int base = 10) {
 
     SAFE_CHAR_BUF(arr, BUF_LEN);
 
-    char *tmp = mpz_get_str(arr, base, t);
+    char *hex = mpz_get_str(arr, 16, t);
 
-    ret = std::string(tmp);
+    ret = hex;
 
-    goto clean;
+    int n_zeroes = numBytes * 2 - ret.length();
+    if (n_zeroes > 0) {
+      ret.insert(0, n_zeroes, '0');
+    }
 
   } catch (std::exception &e) {
     LOG_ERROR(e.what());
-    goto clean;
   } catch (...) {
     LOG_ERROR("Unknown throwable");
-    goto clean;
   }
 
-clean:
   mpz_clear(t);
   return ret;
 }
 
-std::string ConvertG2ElementToString(const libff::alt_bn128_G2 &elem,
-                                     int base = 10,
-                                     const std::string &delim = ":") {
+/**
+ * Converts G2 element to string.
+ * Converts inidivudally each field element to string, and concatenates them.
+ * Total size will always be 256 (64 * 4)
+ */
+std::string G2ToString(libff::alt_bn128_G2 elem) {
+  std::string pkey_str;
 
-  std::string result = "";
+  elem.to_affine_coordinates();
 
-  try {
+  pkey_str += fieldElementToHex(elem.X.c0);
+  pkey_str += fieldElementToHex(elem.X.c1);
+  pkey_str += fieldElementToHex(elem.Y.c0);
+  pkey_str += fieldElementToHex(elem.Y.c1);
 
-    result += fieldElementToString(elem.X.c0);
-    result += delim;
-    result += fieldElementToString(elem.X.c1);
-    result += delim;
-    result += fieldElementToString(elem.Y.c0);
-    result += delim;
-    result += fieldElementToString(elem.Y.c1);
+  return pkey_str;
+}
+
+std::string convertHexToDec(char *hex_str) {
+  mpz_t dec;
+  mpz_init(dec);
 
-    return result;
+  std::string output;
 
+  try {
+    std::string hex(hex_str, 64);
+    if (mpz_set_str(dec, hex.c_str(), 16) == -1) {
+      mpz_clear(dec);
+      LOG_ERROR("Bad formatted hex string provided");
+      return output;
+    }
+
+    char arr[mpz_sizeinbase(dec, 10) + 2];
+    char *tmp = mpz_get_str(arr, 10, dec);
+
+    output = tmp;
   } catch (std::exception &e) {
     LOG_ERROR(e.what());
-    return result;
   } catch (...) {
-    LOG_ERROR("Unknown throwable");
-    return result;
+    LOG_ERROR("Exception in convert hex to dec");
```

Comment:
Same comment as above - how does the caller of the function find out that an error occured?